gprofiler/system_metrics.py

```diff
@@ -30,7 +31,7 @@ def start(self):
     def stop(self):
         self._stop_event.set()
         assert self._thread is not None, "SystemMetricsMonitor was not running"
-        self._thread.join()
+        self._thread.join(STOP_TIMEOUT_SECONDS)
```

Comment:
Please raise if timeout happened. I don't want to leak threads with this.
![image](https://user-images.githubusercontent.com/58514213/124564822-1c610a80-de4a-11eb-97ac-15cc3780cdc0.png)

Reply:
Added